flutter_app/lib/widgets/waypoint_queue_widget.dart

```diff
@@ -0,0 +1,186 @@
+import 'package:flutter/material.dart';
+import 'package:imacs/modules/queue_waypoints.dart';
+
+/// Widget to queue waypoints of a drone using MAVLink communication.
+///
+/// This widget displays a table that contains a queue of waypoints to be
+/// sent to the drone. It provides functionality to add a waypoint to the
+/// queue as well as bypass the queue and send a waypoint to the drone
+/// immediately.
+class WaypointQueue extends StatefulWidget {
+  /// @brief Constructs a WaypointQueue widget.
+  ///
+  /// @param queueWaypoints
+  /// QueueWaypoints class instance
+  ///
+  /// @param systemId
+  /// system ID for command constructor
+  ///
+  /// @param componentId
+  /// component ID for command constructor
+  ///
+  final QueueWaypoints queueWaypoints;
+  final int systemId;
+  final int componentId;
+
+  const WaypointQueue({
+    Key? key,
+    required this.systemId,
+    required this.componentId,
+    required this.queueWaypoints,
+  }) : super(key: key);
+
+  @override
+  WaypointQueueState createState() => WaypointQueueState();
+}
+
+/// State for the QueueWaypoints widget.
+class WaypointQueueState extends State<WaypointQueue> {
+  final TextEditingController _latitudeInput = TextEditingController();
+  final TextEditingController _longitudeInput = TextEditingController();
+  final TextEditingController _altitudeInput = TextEditingController();
+
+  late double _latitude;
+  late double _longitude;
+  late double _altitude;
+
+  /// Gets the waypoints from the TextFields
+  void _getWaypointsFromInput() {
+    try {
+      _latitude = double.parse(_latitudeInput.text);
+      _longitude = double.parse(_longitudeInput.text);
+      _altitude = double.parse(_altitudeInput.text);
+    } catch (e) {
+      print('Enter valid numbers.');
+    }
+  }
+
+  /// Sends a command to queue a waypoint.
+  void _queueWaypoint() {
+    _getWaypointsFromInput();
+    widget.queueWaypoints.queueWaypoint(
+      widget.systemId,
+      widget.componentId,
+      _latitude,
+      _longitude,
+      _altitude,
+    );
+    setState(() {});
+  }
+
+  /// Sends a command to send a waypoint, bypassing the queue.
+  void _queueWaypointWithoutQueue() {
+    _getWaypointsFromInput();
+    widget.queueWaypoints.sendWaypointWithoutQueue(
+      widget.systemId,
+      widget.componentId,
+      _latitude,
+      _longitude,
+      _altitude,
+    );
+  }
+
+  /// Sends the first waypoint in the queue to the drone.
+  void _sendNextWaypointInQueue() {
+    widget.queueWaypoints.sendNextWaypointInQueue();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
```

Comment:
Temporarily wrap this in a `SizedBox` with a fixed height and width for testing and showcase. It will be merged this way but we'll change it when integrating widget together (which is pretty far down the line)